esmvaltool/recipes/recipe_hyint.yml

```diff
@@ -37,6 +38,12 @@ diagnostics:
     scripts:
       main:
         script: hyint/hyint.R
+        # General
+        run_diagnostic: true
```

Comment:
Most of these options should be set in [config-user.yml](https://github.com/ESMValGroup/ESMValTool/blob/version2_development/esmvaltool/config-user.yml) instead of in the recipe.

Reply:
Do you mean run_diagnostic, write_netcdf, write_plots? OK, but something like write_plots is difficult to remove from here. How can a user call two diagnostics requiring only parts of their functionalities? E.g. when needing only the netcdf file from a first diagnostic (which would require write_plots: false) and using a second one to do a further processing with plots (which would require write_plots: true)? This would be the case when running the extremeEvents to produce netcdf with climdex indices (avoiding plots) followed hyint to produce joint analysis of hyint indices and climdex indices. 

Forcing these diagnostic-dependent parameters in a single configuration file is not user friendly. Why not allowing to reassign these parameters in recipes for special cases? The same applies to the path to data.

Reply:
> Do you mean run_diagnostic, write_netcdf, write_plots? 

Yes

> How can a user call two diagnostics requiring only parts of their functionalities? E.g. when needing only the netcdf file from a first diagnostic (which would require write_plots: false) and using a second one to do a further processing with plots (which would require write_plots: true)?

I agree that it could make sense in that case, though I do not think it is a problem if you get some extra plots out of a diagnostic run.

> Forcing these diagnostic-dependent parameters in a single configuration file is not user friendly. Why not allowing to reassign these parameters in recipes for special cases?

I think we need a careful look at these configuration options at some point, i.e. if they are necessary at all, but maybe not now. They add complication to diagnostic scripts and many diagnostic scripts do not even implement them.

The option `run_diagnostic` doesn't make much sense in a recipe, you can just comment out the diagnostic that you do not want to run or if you only want to run some of the diagnostics from a recipe we have a command line option for that.

> The same applies to the path to data.

Which path to the data?

Reply:
> Which path to the data?
OBS in the config-user.yml could be more handy for the local user if specified directly in the recipe as one may have data in different locations

Reply:
The recipe should remain 100% portable, therefore all path must be defined in `config-user.yml`.

Reply:
> one may have data in different locations

Note that it is possible to specify multiple data locations in config-user.yml, e.g.

```yaml
rootpath:
  OBS: [~/obs_inputpath1, ~/obs_inputpath2]
```

The reason that this is done in a single config-user.yml file instead of in the recipe is that we want to be able to run all recipes with just one user configuration file. Otherwise you would need to adapt every recipe to your local paths before you can run it.

Reply:
OK, the multiple path option makes it much more handy, thanks.